zk_dtypes/include/field/finite_field.h

```diff
@@ -52,13 +52,13 @@ class FiniteField {
         static_assert(BasePrimeField::Config::kHasTwoAdicRootOfUnity);
         return ComputeTonelliShanksSquareRoot(*static_cast<const F*>(this));
       }
-    } else if constexpr (F::ExtensionDegree() == 2) {
-      if constexpr (p % 4 == 3) {
+    } else if constexpr (F::ExtensionDegree() % 2 == 0) {
+      if constexpr (IsModulusToHalfDegreeThreeModFour<F>()) {
```

Comment:
can we rename this to `isAlgorithm9Compatible`? I doubt it will be used anywhere else.